src/app/pages/notifications-page/notifications.page.ts

```diff
@@ -50,4 +51,11 @@ export class NotificationPage implements OnInit {
     const icon = notification.iconUrl;
     return icon ? icon : '';
   }
+
+  onDeleteNotification(notification: any) {
+    this.notificationModelService.deleteNotification(notification.id).then((data)=>{
+      this.notificationModelService.init();
```

Comment:
..to clarify, as opposed to having the component which called the service know to initialize the model, the service should know to do that, and it should do it itself. It will always need to be done after a delete, so the component should not be given that responsibility.

Reply:
Would this change be what you mean?